src/libtorch.cc

```diff
@@ -929,6 +907,15 @@ ModelInstanceState::ReadOutputTensors(
         (device_.type() == torch::kCPU) ? TRITONSERVER_MEMORY_CPU
                                         : TRITONSERVER_MEMORY_GPU,
         (device_.type() == torch::kCPU) ? 0 : device_.index());
+
+    // Due to pytorch running inference asynchronously, we need set timestamp
+    // after first output is read to get a more accuracte estimate of compute
+    // infer time. This means the compute output time reported would be smaller
+    // than it is in reality. We do this because synchronizing manually causes a
+    // large drop in performance for fast running models.
+    if (idx == 0) {
+      SET_TIMESTAMP(*compute_end_ns);
```

Comment:
Shouldn't you place this after line 890?
https://github.com/triton-inference-server/pytorch_backend/pull/23/files#diff-859dcc1d94f2d13b0d19e7a14af430cd18eb66395a6f69f2d4886854319c4467L890